File: core/src/ru/myitschool/dcrawler/entities/skills/check/PlayCheck.java
package ru.myitschool.dcrawler.entities.skills.check;

import ru.myitschool.dcrawler.entities.skills.Skill;
import ru.myitschool.dcrawler.entities.skills.Target;

/**
 * Created by Voyager on 19.04.2018.
 */
public abstract class PlayCheck {

    Skill skill;

    public PlayCheck(Skill skill) {
        this.skill = skill;
    }

    public abstract int check(Target target);

    public Skill getSkill() {
        return skill;
    }
}

File: core/src/com/wolg_vlad/dcrawler/entities/skills/check/PlayCheck.kt
package com.wolg_vlad.dcrawler.entities.skills.check

import com.wolg_vlad.dcrawler.entities.skills.Skill
import com.wolg_vlad.dcrawler.entities.skills.Target

/**
 * Class for checks. Returns result of attack at this target.
 */
abstract class PlayCheck(var skill: Skill) {
    abstract fun check(target: Target): Int

}